### PR TITLE
README rework

### PR DESCRIPTION
**Pull request by Shillaker**
Note, this PR is best reviewed by looking at the final version of the docs [on the branch](https://github.com/scaleway/serverless-scaleway-functions/tree/readme-containers-issue).

*What's changed?*

- Move all non-essential docs to `docs` folder
- Show examples of _both_ function and container configuration in `README`
- Show configuration in configuration file format, not bullet points in text
- Remove `docs/README.md`
- Fixed version of `prettier` used in Linting (`3.0.0` is latest and finds issues all over codebase)

*Why is this necessary?*

- `README` was long, verbose and inaccurate in parts
- `README` made it difficult to differentiate between function and container config options
- Putting the config file in text bullet points was hard to translate to YAML. Easier to copy-paste from a YAML format
- `docs/README.md` was almost all duplicated information
- A single large markdown file is difficult to browse from a terminal, separate files in a `docs` folder are easier

### CODE REVIEW

**Comment by asuri0n** [comment]
good updates :+1:

File: README.md [hunk 1]
```diff
@@ -1,579 +1,283 @@
...
-  # Global Environment variables - used in every functions
+
+  # Global environment variables, used in every function
   env:
-    test: test
-  # Storing credentials in this file is strongly not recommanded for security concerns, please refer to README.md about best practices
+    MY_VAR: "some var"
+    MY_OTHER_VAR: "some other var"
+
+  # Global secrets, used in every function
```
Comment (thomas-tacquet) [inline README.md:106]:
```suggestion
  # Global secrets available for Container / Functions in this namespace
```

File: README.md [hunk 2]
```diff
@@ -1,579 +1,283 @@
...
 provider:
+  # Must not change
   name: scaleway
+
+  # Function runtime
+  # List: https://www.scaleway.com/en/docs/serverless/functions/reference-content/functions-lifecycle/#available-runtimes
   runtime: python310
-  # Global Environment variables - used in every functions
+
+  # Global environment variables, used in every function
```
Comment (thomas-tacquet) [inline README.md:101]:
```suggestion
  # Global environment variables, available for Container / Functions in this namespace
```

File: README.md [hunk 3]
```diff
@@ -1,579 +1,283 @@
...
-Node has two module systems: `CommonJS` modules and `ECMAScript` (`ES`) modules. By default, Node treats your code files as CommonJS modules, however [ES modules](https://nodejs.org/api/esm.html) have also been available since the release of `node16` runtime on Scaleway Serverless Functions. ES modules give you a more modern way to re-use your code.
-
-According to the official documentation, to use ES modules you can specify the module type in `package.json`, as in the following example:
-
-```json
-  ...
-  "type": "module",
-  ...
-```
+    # Minimum and maximum number of instances of the function
```
Comment (thomas-tacquet) [inline README.md:131]:
```suggestion
    # Minimum and maximum number of instances
```

File: README.md [hunk 4]
```diff
@@ -1,579 +1,283 @@
...
-  runtime: go118
-functions:
-  main:
-    handler: "."
-  testing:
-    handler: src/testing
-  second:
-    handler: src/second
-```
+    # Memory limit for the function (in MiB)
```
Comment (thomas-tacquet) [inline README.md:135]:
```suggestion
    # Memory limit (in MiB)
```

File: README.md [hunk 5]
```diff
@@ -1,579 +1,283 @@
...
-Recommended folder structure for `php` runtimes:
+    # Runtime for this function, allows overriding provider.runtime
+    runtime: node20
 
-```yml
-├── handler.php
-├── composer.json (not necessary if you do not need dependencies)
-└── serverless.yml
-```
+    # Whether to permit HTTP or HTTPS-only. Options: enabled, disabled, or redirected
```
Comment (thomas-tacquet) [inline README.md:146]:
```suggestion
    # Whether to permit HTTP or HTTPS-only. Options: enabled, disabled (https only), or redirected (http -> https)
```

File: README.md [hunk 6]
```diff
@@ -1,579 +1,283 @@
...
+    httpOption: enabled
 
-Your serverless.yml `functions` should look something like this:
+    # Whether the container is public (no authentication), or private (uses token-based authentication)
```
Comment (thomas-tacquet) [inline README.md:149]:
```suggestion
    # Controls the privacy: public (no authentication), or private (uses token-based authentication)
```

File: README.md [hunk 7]
```diff
@@ -1,579 +1,283 @@
...
+    events:
+      - schedule:
+          rate: "1 * * * *"
+          # Data passed as input in the request
+          input:
+            key-a: "value-a"
+            key-b: "value-b"
+
+# Define containers, cannot be used with functions
+custom:
```
Comment (thomas-tacquet) [inline README.md:175]: maybe separate both files ? It can be impressive to see a huge configuration file

**Review by thomas-tacquet — APPROVED** [review]

**Pull request merged**